File: reading-backend-node/src/utils/fileUtil.js
import { randomUUID } from 'crypto';
import { BusinessError } from '../errors.js';
import { ResultCode } from './result.js';

const ALLOWED_EXTENSIONS = ['jpg', 'jpeg', 'png', 'gif', 'webp'];
const MAX_FILE_SIZE = 5 * 1024 * 1024;

export function validateImage(file) {
  if (!file || !file.size) {
    throw new BusinessError('上传文件不能为空');
  }
  if (file.size > MAX_FILE_SIZE) {
    throw new BusinessError('文件大小不能超过5MB');
  }
  const originalFilename = file.originalname;
  if (!originalFilename || !originalFilename.includes('.')) {
    throw new BusinessError(ResultCode.BAD_REQUEST.code, '文件格式不正确');
  }
  const extension = originalFilename.substring(originalFilename.lastIndexOf('.') + 1).toLowerCase();
  if (!ALLOWED_EXTENSIONS.includes(extension)) {
    throw new BusinessError(ResultCode.BAD_REQUEST.code, '仅支持 jpg、jpeg、png、gif、webp 格式');
  }
}

export function generateFilename(originalFilename) {
  const extension = originalFilename.substring(originalFilename.lastIndexOf('.') + 1).toLowerCase();
  return randomUUID().replace(/-/g, '') + '.' + extension;
}

File: reading-backend/util/FileUtil.java
package com.example.reading.util;

import com.example.reading.common.ResultCode;
import com.example.reading.exception.BusinessException;
import org.springframework.web.multipart.MultipartFile;

import java.util.Arrays;
import java.util.List;
import java.util.UUID;

public final class FileUtil {

    private static final List<String> ALLOWED_EXTENSIONS = Arrays.asList("jpg", "jpeg", "png", "gif", "webp");
    private static final long MAX_FILE_SIZE = 5 * 1024 * 1024;

    private FileUtil() {
    }

    public static void validateImage(MultipartFile file) {
        if (file == null || file.isEmpty()) {
            throw new BusinessException("上传文件不能为空");
        }
        if (file.getSize() > MAX_FILE_SIZE) {
            throw new BusinessException("文件大小不能超过5MB");
        }
        String originalFilename = file.getOriginalFilename();
        if (originalFilename == null || !originalFilename.contains(".")) {
            throw new BusinessException(ResultCode.BAD_REQUEST, "文件格式不正确");
        }
        String extension = originalFilename.substring(originalFilename.lastIndexOf('.') + 1).toLowerCase();
        if (!ALLOWED_EXTENSIONS.contains(extension)) {
            throw new BusinessException(ResultCode.BAD_REQUEST, "仅支持 jpg、jpeg、png、gif、webp 格式");
        }
    }

    public static String generateFilename(String originalFilename) {
        String extension = originalFilename.substring(originalFilename.lastIndexOf('.') + 1).toLowerCase();
        return UUID.randomUUID().toString().replace("-", "") + "." + extension;
    }
}
